src/main/java/uk/gov/hmcts/divorce/payment/PaymentStatusService.java

```diff
@@ -58,16 +66,34 @@ public void hasSuccessFulPayment(List<uk.gov.hmcts.reform.ccd.client.model.CaseD
                 .toList()
         );
 
-        final String userToken = idamService.retrieveSystemUpdateUserDetails().getAuthToken();
+        final User user = idamService.retrieveSystemUpdateUserDetails();
+        final String userToken = user.getAuthToken();
         final String s2sToken = authTokenGenerator.generate();
 
-        final List<Long> caseIds = casesWithInProgressPayments
+        final List<Long> caseIdsWithSuccessfulPayment = casesWithInProgressPayments
             .parallelStream()
             .filter(caseDetails -> hasSuccessfulPayment(caseDetails, userToken, s2sToken))
             .map(CaseDetails::getId)
             .toList();
 
-        log.info("PaymentStatusService found with successful payments: " + caseIds);
+        triggerPaymentMadeEvent(casesWithInProgressPayments, user, s2sToken, caseIdsWithSuccessfulPayment);
+
+        log.info("PaymentStatusService found with successful payments: " + caseIdsWithSuccessfulPayment);
+    }
+
+    private void triggerPaymentMadeEvent(List<CaseDetails<CaseData, State>> casesInProgressPayment, User user,  String s2sToken,
+                                         List<Long> caseIdsWithSuccessfulPayment) {
+        casesInProgressPayment.stream().filter(cases -> caseIdsWithSuccessfulPayment.contains(cases.getId())).forEach(successfulPaymentCase -> {
```

Comment:
We could remove `.map(CaseDetails::getId)` from line 76 and pass `casesWithSuccessfulPayment` directly into this method to save searching the successful case IDs for each case.

Or alternatively, collect the IDs into a set and then `.contains` in this method will be able to lookup the ID in 1 step rather than looping over the list. I think it would be simpler to pass the successful into this method though and then map to IDs before the logging is done on line 81

Reply:
Refactored code as suggested